### PR TITLE
Update functions used for testing to be v2

### PR DESCRIPTION
**Pull request by rlazo**
Functions v1 have been deprecated for a while, and we should use v2 when writing functions.

### CODE REVIEW

File: functions/functions/package.json [hunk 1]
```diff
@@ -2,13 +2,13 @@
   "name": "functions",
   "description": "Cloud Functions for Android Functions Quickstart",
   "engines": {
-    "node": "10"
+    "node": "22"
```
Comment (thatfiredev) [inline functions/functions/package.json:5]: Most Firebase repos use either Node 18 or 20, as we want to keep compatibility for developers in older versions.

Node 18 will reach EoL on April 2025 ([source](https://nodejs.org/en/about/previous-releases)) - should we go with 20 instead?

**Review by thatfiredev — CHANGES_REQUESTED** [review]
Thanks for updating to the modular Admin SDK too :D

Just one minor change needed before we can merge

File: functions/functions/package.json (hunk 1 above)
Comment (rlazo) [reply functions/functions/package.json:5]: Sure, can go with that :-)

**Review by thatfiredev — APPROVED** [review]
Thanks @rlazo !

**Pull request merged**